### DISCUSSION TITLE
Dual upscaler with weight

### DISCUSSION THREAD

**Discussion by Alyndiar** [vladmandic/sdnext#262] [Ideas]
I was wondering. In Auto1111, the upscale section lets you select 2 upscaler with a weight to balance how much each upscaler influences the final result. I notice that this is not present in your fork. Would it be possible to get this feature back, maybe with an option to enable and disable it^ I was using this mix of upscalers to get exactly the sharpness and details I wanted without introducing too many extra artefacts.

**Comment by vladmandic** [comment]
In my experience, 99% of use-cases are better suited by choosing whatever upscaler is best for the job, not mixing multiple upscalers.
Yes, mixing upscalers seems cool, but it adds a lot of extra options to UI that majority of users don't care about or don't even understand.

Btw, did you try finding/installing additional upscalers that would do a better job? Good thing about ESRGAN is that its pluggable.
Upscalers such as `4x-Valar` or `4x-UltraSharp` are compatible and perform pretty well.

**Reply by Alyndiar** [reply]
Yes, I regularly use 4x-Valar and 4x-UltraSharp, but by themselves, they introduce a lot of defects in the final image. I find using Swim4x with 25%-50% 4x-Valar of 4x-Ultrasharp gives a much better result nearly all the time. You get a good sharpening from the 4x-... upscalers but mixing with Swim4x keeps everything more consistent with the original image and reduces upscaling and sharpening artefacts.

I totally understand if you don't want to make multiple upscalers available as I guess most user effectively don't care about it that much, but I did have to ask. I'm just sad since this is the only thing missing for me to completely drop the original Automatic1111 WebUI and switch to your fork once and for all. Everything else works better for me with your version. I'd do it myself if I could, I just don't know enough Python to figure it out.
